test/rubygems/test_gem_specification.rb

```diff
@@ -3951,6 +3951,34 @@ def test_load_default_gem
     assert_equal ["default-2.0.0.0"], Gem::Specification.map(&:full_name)
   end
 
+  def test_validate_for_resolution_validates_required_attributes
+    e = assert_raise Gem::InvalidSpecificationException do
+      @a1.version = nil
+      @a1.validate_for_resolution
+    end
+
+    assert_equal "missing value for attribute version", e.message
+  end
+
+  def test_validate_for_resolution_validates_name
+    e = assert_raise Gem::InvalidSpecificationException do
+      @a1.name = 123
+      @a1.validate_for_resolution
+    end
+
+    assert_equal 'invalid value for attribute name: "123" must be a string', e.message
+  end
+
+  def test_validate_for_resolution_validates_duplicate_dependencies
+    e = assert_raise Gem::InvalidSpecificationException do
+      @a1.add_dependency "foo", "1.2.3"
+      @a1.add_dependency "foo", "3.4.5"
+      @a1.validate_for_resolution
+    end
+
+    assert_match "duplicate dependency on foo", e.message
+  end
+
```

Comment:
I didn't add a test case for platform validations because I couldn't figure out how to set up a spec with an invalid platform—`Specification#platform=` doesn't allow it.

Reply:
I guess it could be set directly with `instance_variable_set`. People do all short of things for all sort of reasons so I guess having some validation doesn't hurt.